### PR TITLE
Transform tf_robust to tensorgraph

### PR DESCRIPTION
**Pull request by miaecle**
Working on implementing tensorgraph style robust multitask network and progressive network.

### CODE REVIEW

**Comment by rbharath** [comment]
@miaecle Awesome to see this! Just merged in #1082 which removes the `tensorflow_models` folder. You should be able to fix the merge conflict by moving over your models to the `tensorgraph` folder. Sorry for the extra work.

**Comment by miaecle** [comment]
@rbharath No problem! Just got it fixed, progressive network seems like some work, let's merge the robust multitask first.

File: deepchem/molnet/__init__.py [hunk 1]
```diff
@@ -33,4 +33,4 @@
 from deepchem.molnet.dnasim import simulate_single_motif_detection
 
 from deepchem.molnet.run_benchmark import run_benchmark
-from deepchem.molnet.run_benchmark_low_data import run_benchmark_low_data
+#from deepchem.molnet.run_benchmark_low_data import run_benchmark_low_data
```
Comment (miaecle) [inline deepchem/molnet/__init__.py]: Since low data models are moved to contrib, I temporarily commented out these parts in molnet, could do modifications once low data models are back.

File: deepchem/models/tensorgraph/progressive_multitask.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        layer = layer + lateral_contrib
+      outputs.append(layer)
+      self.add_output(layer)
+      task_label = Label(shape=(None, 1))
+      task_weight = Weights(shape=(None, 1))
+      weighted_loss = ReduceSum(
+          L2Loss(in_layers=[task_label, layer, task_weight]))
+      self.create_submodel(
+          layers=task_layers, loss=weighted_loss, optimizer=None)
+    # Weight decay not activated
```
Comment (miaecle) [inline deepchem/models/tensorgraph/progressive_multitask.py]: Weight decay seems to be deactivated in the original implementation. I could not find a good way to merge in this function as well, any good idea?

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 3]
```diff
@@ -928,6 +928,7 @@ def get_train_op(self):
         optimizer = self.graph.optimizer
       else:
         optimizer = self.optimizer
+      # Should we keep a separate global step count for each submodel?
```
Comment (miaecle) [inline deepchem/models/tensorgraph/tensor_graph.py]: For progressive model, using a shared global step doesn't seem natural(in case of learning rate decay), defining a separate parameter might be better.

File: deepchem/models/tests/test_overfit.py [hunk 4]
```diff
@@ -936,16 +895,12 @@ def test_tf_progressive_regression_overfit(self):
         dropouts=[0.],
         learning_rate=0.003,
         weight_init_stddevs=[.1],
-        seed=123,
         alpha_init_stddevs=[.02],
         batch_size=n_samples)
 
     # Fit trained model
     model.fit(dataset, nb_epoch=20)
-    model.save()
```
Comment (miaecle) [inline deepchem/models/tests/test_overfit.py]: Need to mention that progressive model currently cannot be saved, still trying to dig into it but I suspect the submodel setting might be the cause.

File: deepchem/models/tensorgraph/progressive_multitask.py (hunk 2 above)
Comment (rbharath) [reply deepchem/models/tensorgraph/progressive_multitask.py]: I think it should be feasible to to just add on a new loss term manually if needed. That said, I don't believe weight decay made a big difference so I'm fine to proceed without it.

File: deepchem/models/tests/test_overfit.py (hunk 4 above)
Comment (rbharath) [reply deepchem/models/tests/test_overfit.py]: @miaecle Would you mind raising a new issue documenting the error? This sounds like a general problem we should fix. We can do this in a different PR though.

**Comment by rbharath** [comment]
@miaecle This is great work! Thanks for taking on the conversion task :)

I added a couple comments, but I think these are all fine to address in different PRs.

LGTM
